src/Analyzers/CSharp/Analyzers/ConvertSwitchStatementToExpression/ConvertSwitchStatementToExpressionDiagnosticAnalyzer.Analyzer.cs

```diff
@@ -3,29 +3,38 @@
 // See the LICENSE file in the project root for more information.
 
 using System;
+using System.Diagnostics;
 using System.Linq;
+using Microsoft.CodeAnalysis;
 using Microsoft.CodeAnalysis.CSharp.Extensions;
 using Microsoft.CodeAnalysis.CSharp.Syntax;
 using Microsoft.CodeAnalysis.Shared.Extensions;
 
 namespace Microsoft.CodeAnalysis.CSharp.ConvertSwitchStatementToExpression
 {
+    using static ConvertSwitchStatementToExpressionHelpers;
+
     internal sealed partial class ConvertSwitchStatementToExpressionDiagnosticAnalyzer
     {
         private sealed class Analyzer : CSharpSyntaxVisitor<SyntaxKind>
         {
+            private readonly bool _supportsOrPatterns;
+
             private ExpressionSyntax _assignmentTargetOpt;
 
-            private Analyzer()
+            private Analyzer(bool supportsOrPatterns)
             {
+                _supportsOrPatterns = supportsOrPatterns;
             }
 
             public static (SyntaxKind nodeToGenerate, VariableDeclaratorSyntax declaratorToRemoveOpt) Analyze(
                 SwitchStatementSyntax node,
                 SemanticModel semanticModel,
                 out bool shouldRemoveNextStatement)
             {
-                var analyzer = new Analyzer();
+                var parseOptions = (CSharpParseOptions)semanticModel.SyntaxTree.Options;
+                // TODO: use CSharp9 version once available.
```

Comment:
Tracking issue?

Reply:
tracking issue is: https://github.com/dotnet/roslyn/issues/43348